src/response.rs

```diff
@@ -202,6 +201,51 @@ impl Response {
         self.cookie_events.push(CookieEvent::Removed(cookie));
     }
 
+    /// Returns an optional reference to an error if the response contains one.
+    pub fn error(&self) -> Option<&Error> {
+        self.res.error()
+    }
+
+    /// Returns an optional reference to a downcasted error if the response contains an error with a matching typeid.
```

Comment:
```suggestion
    /// Returns a reference to the original error associated with this response if there is one and it can be downcast to the specified type.
```